### ISSUE TITLE
The pub(crate) structure can add the method or field of pub

### ISSUE THREAD

**Issue by hi-rustin** [rust-lang/rust#85245]
I tried this code:

```rust
pub(crate) struct Foo {
    pub(crate) a: i32,
    pub b: i32,
}

impl Foo {
    pub fn get_a(&self) -> i32 {
        self.a
    }

    pub(crate) fn get_b(&self) -> i32 {
        self.b
    }
}


fn main() {
    let foo = Foo {
        a: 1,
        b:2,
    };
    println!("{}",foo.get_a());
    println!("{}",foo.get_b());
}
```
For example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=03625034caf45a647e896523c3f861c2

I expected to see this happen: 

Pub fields and methods should be warned.

Instead, this happened: 

There are no warnings. cargo clippy does not warn either.

### Meta

`rustc --version --verbose`:
```
rustc --version --verbose
rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-apple-darwin
release: 1.52.1
LLVM version: 12.0.0
<version>
```

<details><summary>Backtrace</summary>
<p>

```
<backtrace>
```

</p>
</details>

**Comment by ehuss** [comment]
There is the [unreachable-pub](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unreachable-pub) lint which will issue warnings for the `pub` fields/methods, is that what you are looking for?

**Comment by hi-rustin** [comment]
> There is the [unreachable-pub](https://doc.rust-lang.org/rustc/lints/listing/allowed-by-default.html#unreachable-pub) lint which will issue warnings for the `pub` fields/methods, is that what you are looking for?

Yes. Thanks!
